refinery/core/api.py

```diff
@@ -282,6 +282,12 @@ def obj_get(self, bundle, **kwargs):
         if not user.is_authenticated() and res and not res.is_public():
             return HttpUnauthorized()
 
+        if not res.is_valid():
+            raise NotFound(
+                "DataSet with UUID: {} is invalid, and most likely is "
```

Comment:
This is `SharableResourceAPIInterface` class. Why is there a reference to DataSet here?